main.py

```diff
@@ -1,29 +1,99 @@
+from report_generator import ReportGenerator
+
+import argparse
 
-CURRENT_DIRECTORY = os.getcwd()
 
 if __name__ == "__main__":
 
-    if len(sys.argv) < 4:
-        print("You must specify files path and the command with proper flag")
-        sys.exit(0)
+    arg_parser = argparse.ArgumentParser(
+        description="Generates reports from the weather data files"
+    )
+
+    arg_parser.add_argument(
+        "data_folder",
+        type=str,
+        action='store',
+        nargs=1,
+        metavar="path/to/folder",
+        help="Folder path where data files are stored"
+    )
+
+    arg_parser.add_argument(
+        '-e',
+        type=validate_year,
+        metavar="year",
+        help="To display the highest, lowest "
+             "temperature and highest humidity"
+    )
+
+    arg_parser.add_argument(
+        '-a',
+        type=validate_year_month,
+        metavar="year/mm",
+        help="To display the average high, low "
+             "temperature and average mean humidity"
+    )
+
+    arg_parser.add_argument(
+        '-c',
+        type=validate_year_month,
+        metavar="year/mm",
+        help="Display charts of high and low "
+             "temperature for each day"
+    )
+
+    arg_parser.add_argument(
+        '-cs',
+        type=validate_year_month,
+        metavar="year/mm",
+        help="Display one line charts of high and low "
+             "temperature for each day"
+    )
+
+    args = arg_parser.parse_args()
+
+    weather_data_folder_path = CURRENT_DIRECTORY + args.data_folder[0]
+    check_directory_path_exists(weather_data_folder_path)
+
+    report_generator = ReportGenerator()
+
+    if args.e:
+        data_provider = ReportDataProvider(args.e, weather_data_folder_path)
+        report_data = data_provider.get_high_low_temp_high_humidity_data()
+
+        report_generator.gen_report_highest_lowest_temp_and_humidity(
+            report_data
+        )
 
-    weather_data_files_path = CURRENT_DIRECTORY + sys.argv[1]
-    if not os.path.isdir(weather_data_files_path):
-        print("Invalid path to directory : ", weather_data_files_path)
-        sys.exit(0)
+    if args.a:
+        data_provider = ReportDataProvider(args.a, weather_data_folder_path)
+        report_data = data_provider.get_average_temp_and_humidity()
 
-    print(weather_data_files_path)
+        report_generator.gen_report_average_max_min_temp_and_mean_humidity(
+            report_data
+        )
 
-    system_args = sys.argv[2:]
+    if args.c:
+        data_provider = ReportDataProvider(args.c, weather_data_folder_path)
+        report_data = data_provider.get_chart_data()
 
-    year = sys.argv[3]
+        report_generator.gen_report_high_low_temperature_charts(
+            report_data
+        )
 
-    file_path_matcher = FilePathMatcherWithDate(weather_data_files_path)
-    file_path_matcher.setDate(year)
-    matched_file_paths_with_year = file_path_matcher.get_files_path()
+    if args.cs:
```

Comment:
args `c` and `cs` are identical till calling `get_chart_data`, this logic should be combined, e.g make a list `[args.c, args.cs]` and loop over it and check if the value is not None then do the rest.

Reply:
like this:
```python

for value in [args.c, args.cs]:

     if value is not None:
           data_provider = ReportDataProvider( value , weather_data_folder_path )
           report_data = data_provider.get_chart_data()
    
          if args.c == value:
                report_data.gen_report_high_low_temperature_charts( report_data)
          if args.cs == value:
               report_data.gen_report_high_low_temperature_single_line_chart( report_data )
   

```